### PR TITLE
chore: bump sdk to current

### PR DESCRIPTION
**Pull request by czarcas7ic**
Closes: #XXX

## What is the purpose of the change

Ensuring tests still pass with some of the changes we made to our fork.

1. We can no longer expect logs in the response and need to parse the response itself https://github.com/osmosis-labs/cosmos-sdk/commit/99c0a54c54a0b17c2ab8893f053759e8185623ec
2. We can no longer expect validator distribution on every block, and need to take into account that we now distribute every 50 blocks https://github.com/osmosis-labs/cosmos-sdk/commit/3d440de123e416f111d7bdd3126b421d1c4f391d

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -397,7 +397,7 @@ replace (
 
 	// Our cosmos-sdk branch is: https://github.com/osmosis-labs/cosmos-sdk/tree/osmo-v23/v0.47.5-iavl-v1, current branch: osmo-v23/v0.47.5-iavl-v1. Direct commit link: https://github.com/osmosis-labs/cosmos-sdk/commit/d44a3847e089de47cbba0f31a0cebed64ad2806f
 	// https://github.com/osmosis-labs/cosmos-sdk/releases/tag/v0.47.5-v23-osmo-1
-	github.com/cosmos/cosmos-sdk => github.com/osmosis-labs/cosmos-sdk v0.47.5-v23-osmo-2-iavl-v1
+	github.com/cosmos/cosmos-sdk => github.com/osmosis-labs/cosmos-sdk v0.47.6-0.20240227001520-a836ab6f5815
```
Comment (mattverse) [inline go.mod:400]: Do we not need to bump it to this commit? https://github.com/osmosis-labs/cosmos-sdk/commit/10406e3b6d6ed3632d66847a08e8480c4e81015a

Comment (czarcas7ic) [reply go.mod:400]: That is the next PR, where I bump prometheus here in osmosis. I wanted to separate the changes

Comment (czarcas7ic) [reply go.mod:400]: The PR is open but in draft, it is what alerted me to the fact that osmosis was behind and failing tests

Comment (mattverse) [reply go.mod:400]: gotcha. Thanks for the explanation!

**Review by mattverse — APPROVED** [review]
LGTM

**Pull request merged**